cli/tests/test_upgrade_public_resolver_bootstrap.py

```
@@ -0,0 +1,396 @@
+        patch.dict(os.environ, {}, clear=True),
+        patch.object(upgrade_module, "_fetch_latest_version") as latest,
+    ):
+        upgrade_module._maybe_delegate_public_upgrade(["upgrade", "--help"])
+    latest.assert_not_called()
+
+
+def test_authentication_uses_exact_identity_digest_marker_and_syntax() -> None:
+    resolver = _resolver_payload()
+    bash = SimpleNamespace(path="/bin/bash", assert_stable=Mock())
+
+    def run_command(argv: list[str], **_kwargs):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `run_command`

(ANN202)